File: sketches/sketches.model/src/main/java/edu/kit/kastel/lissa/sketches/model/types/SketchBoxTypeMapping.java
package edu.kit.kastel.lissa.sketches.model.types;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import edu.kit.kastel.lissa.sketches.model.elements.ISketchElement;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@Documented
public @interface SketchBoxTypeMapping {
    SketchBoxTypes type();

    Class<? extends ISketchElement> implementation();
}

File: sketches/model/src/main/kotlin/edu/kit/kastel/lissa/sketches/model/types/SketchBoxTypeMapping.kt
package edu.kit.kastel.lissa.sketches.model.types

import edu.kit.kastel.lissa.sketches.model.elements.ISketchElement
import kotlin.reflect.KClass

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
@MustBeDocumented
annotation class SketchBoxTypeMapping(
    val type: SketchBoxTypes,
    val implementation: KClass<out ISketchElement>
)
